cmd/watch.go

```diff
@@ -48,9 +48,14 @@ var watchCmd = &cobra.Command{
 			componentName = args[0]
 		}
 
-		_, sourcePath, err := component.GetComponentSource(client, componentName, applicationName, projectName)
+		sourceType, sourcePath, err := component.GetComponentSource(client, componentName, applicationName, projectName)
 		checkError(err, "Unable to get source for %s component.", componentName)
 
+		if sourceType != "binary" && sourcePath != "local" {
```

Comment:
❗️ ❗️ ❗️ 
I think you wanted:
`sourceType != "binary" && sourceType != "local"`
not:
`sourceType != "binary" && sourcePath != "local"`

Reply:
https://github.com/redhat-developer/odo/issues/751

Reply:
Ah! Sorry my bad

Reply:
That happens, no worries.

What is disappointing is that no one bothered to run and test this manually, not one of 3 people 
 that reviewed and approved this and apparently not even you who wrote it :-(

Reply:
Sorry again. I tested locally but only some of the negative cases as it was a small fix. Will be careful in the future.